symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/signal/SignalService.java

```diff
@@ -0,0 +1,163 @@
+
+  public SignalService(SignalsApi signalsApi, AuthSession authSession, RetryWithRecoveryBuilder<?> retryBuilder) {
+    this.signalsApi = signalsApi;
+    this.authSession = authSession;
+    this.retryBuilder = retryBuilder;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public OboSignalService obo(AuthSession oboSession) {
+    return new SignalService(signalsApi, oboSession, retryBuilder);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public List<Signal> listSignals(Integer skip, Integer limit) {
+    return executeAndRetry("listSignals",
+        () -> signalsApi.v1SignalsListGet(authSession.getSessionToken(), authSession.getKeyManagerToken(), skip, limit));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Stream<Signal> listSignalsStream(Integer chunkSize, Integer totalSize) {
+    PaginatedApi<Signal> api = ((this::listSignals));
+
+    final int actualChunkSize = chunkSize == null ? 50 : chunkSize;
+    final int actualTotalSize = totalSize == null ? 50 : totalSize;
+
+    return new PaginatedService<>(api, actualChunkSize, actualTotalSize).stream();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Signal getSignal(String id) {
+    return executeAndRetry("getSignal",
+        () -> signalsApi.v1SignalsIdGetGet(authSession.getSessionToken(), id, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Signal createSignal(BaseSignal signal) {
+    return executeAndRetry("createSignal",
+        () -> signalsApi.v1SignalsCreatePost(authSession.getSessionToken(), signal, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Signal updateSignal(String id, BaseSignal signal) {
+    return executeAndRetry("updateSignal",
+        () -> signalsApi.v1SignalsIdUpdatePost(authSession.getSessionToken(), id, signal, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void deleteSignal(String id) {
+    executeAndRetry("deleteSignal",
+        () -> signalsApi.v1SignalsIdDeletePost(authSession.getSessionToken(), id, authSession.getKeyManagerToken()));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ChannelSubscriptionResponse subscribeSignal(String id, Boolean pushed, List<Long> userIds) {
+    return executeAndRetry("subscribeSignal",
+        () -> signalsApi.v1SignalsIdSubscribePost(authSession.getSessionToken(), id, authSession.getKeyManagerToken(), pushed, userIds));
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public ChannelSubscriptionResponse unsubscribeSignal(String id, List<Long> userIds) {
+    return executeAndRetry("unsubscribeSignal",
+        () -> signalsApi.v1SignalsIdUnsubscribePost(authSession.getSessionToken(), id, authSession.getKeyManagerToken(), userIds));
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public List<ChannelSubscriber> subscribers(String id, Integer skip, Integer limit) {
```

Comment:
is it normal that we have different style of pagination?

Reply:
I think this one is not really a pagination, just a basic call to the endpoint. In this ticket [APP-3183](https://perzoinc.atlassian.net/browse/APP-3183), we also want to expose the skip and limit param in out apis.